lib/irc/ClientPool.js

```diff
@@ -4,11 +4,12 @@
  * and may be closed for a variety of reasons.
  */
 "use strict";
-var stats = require("../config/stats");
-var log = require("../logging").get("ClientPool");
-var Promise = require("bluebird");
+const stats = require("../config/stats");
+const log = require("../logging").get("ClientPool");
+const Promise = require("bluebird");
+const QueuePool = require("../util/QueuePool");
 
-const RECONNECT_TIME_MS = 1000;
+const RECONNECT_CLIENT_QUEUE_SIZE = 20;
```

Comment:
This means we are limited to 20 concurrent ongoing reconnections, not a queue length of 20. Maybe `QUEUE_POOL_SIZE` would better describe this.

Reply:
``CONCURRENT_RECONNECT_LIMIT`` sounds okay?